core/embed/extmod/modtrezorio/modtrezorio-flash.h

```diff
@@ -118,3 +123,302 @@ STATIC const mp_obj_type_t mod_trezorio_FlashOTP_type = {
+
+/// if __debug__:
+///     def erase(self) -> None:
+///         """
+///         Erases the whole flash area.
+///         """
+STATIC mp_obj_t mod_trezorio_FlashArea_erase(mp_obj_t obj_self) {
+  mp_obj_FlashArea_t *self = (mp_obj_FlashArea_t *)MP_OBJ_TO_PTR(obj_self);
+  if (sectrue != flash_area_erase(self->area, NULL)) {
+    mp_raise_ValueError("Erase failed.");
+  }
+  return mp_const_none;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorio_FlashArea_erase_obj,
+                                 mod_trezorio_FlashArea_erase);
+
+#endif  // DEBUG
+
+STATIC const mp_rom_map_elem_t mod_trezorio_FlashArea_locals_dict_table[] = {
+    {MP_ROM_QSTR(MP_QSTR_size), MP_ROM_PTR(&mod_trezorio_FlashArea_size_obj)},
+    {MP_ROM_QSTR(MP_QSTR_hash), MP_ROM_PTR(&mod_trezorio_FlashArea_hash_obj)},
+#if DEBUG
+    {MP_ROM_QSTR(MP_QSTR_read), MP_ROM_PTR(&mod_trezorio_FlashArea_read_obj)},
+    {MP_ROM_QSTR(MP_QSTR_write), MP_ROM_PTR(&mod_trezorio_FlashArea_write_obj)},
+    {MP_ROM_QSTR(MP_QSTR_erase_sector),
+     MP_ROM_PTR(&mod_trezorio_FlashArea_erase_sector_obj)},
+    {MP_ROM_QSTR(MP_QSTR_erase), MP_ROM_PTR(&mod_trezorio_FlashArea_erase_obj)},
+#endif
+};
+
+STATIC MP_DEFINE_CONST_DICT(mod_trezorio_FlashArea_locals_dict,
+                            mod_trezorio_FlashArea_locals_dict_table);
+
+STATIC const mp_obj_type_t mod_trezorio_FlashArea_type = {
+    {&mp_type_type},
+    .name = MP_QSTR_FlashArea,
+    .locals_dict = (void *)&mod_trezorio_FlashArea_locals_dict,
+};
+
+#define FLASH_AREA(name, area_id)                                    \
+  STATIC const mp_obj_FlashArea_t mod_trezorio_flash_area_##name = { \
+      .base = {.type = MP_ROM_PTR(&mod_trezorio_FlashArea_type)},    \
+      .area = &area_id,                                              \
+  };
+
+/// mock:global
+/// package: trezorio.flash_area
+/// from . import FlashArea
+/// BOARDLOADER: FlashArea
+/// BOOTLOADER: FlashArea
+/// FIRMWARE: FlashArea
+/// STORAGE_A: FlashArea
+/// STORAGE_B: FlashArea
+/// TRANSLATIONS: FlashArea
+/// SECRET: FlashArea
+FLASH_AREA(BOARDLOADER, BOARDLOADER_AREA)
+FLASH_AREA(BOOTLOADER, BOOTLOADER_AREA)
+FLASH_AREA(FIRMWARE, FIRMWARE_AREA)
+FLASH_AREA(STORAGE_A, STORAGE_AREAS[0])
+FLASH_AREA(STORAGE_B, STORAGE_AREAS[1])
+FLASH_AREA(TRANSLATIONS, TRANSLATIONS_AREA)
+#if USE_OPTIGA && !defined(TREZOR_EMULATOR)
+FLASH_AREA(SECRET, SECRET_AREA)
+#endif
+
+#define MP_ROM_FLASH_AREA(name) \
+  { MP_ROM_QSTR(MP_QSTR_##name), MP_ROM_PTR(&mod_trezorio_flash_area_##name) }
+
+STATIC const mp_rom_map_elem_t mod_trezorio_flash_area_globals_table[] = {
+    {MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_flash_area)},
+    MP_ROM_FLASH_AREA(BOARDLOADER),
+    MP_ROM_FLASH_AREA(BOOTLOADER),
+    MP_ROM_FLASH_AREA(FIRMWARE),
+    MP_ROM_FLASH_AREA(STORAGE_A),
+    MP_ROM_FLASH_AREA(STORAGE_B),
+    MP_ROM_FLASH_AREA(TRANSLATIONS),
+#if USE_OPTIGA && !defined(TREZOR_EMULATOR)
+    MP_ROM_FLASH_AREA(SECRET),
+#endif
```

Comment:
TODO: in production mode, we should also not create objects for storage / secret

Reply:
on U5 models, SECRET area is inaccessible to FW even in debug, so maybe that shouldn't be created there regardless on `PRODUCTION`

Reply:
e442983a3ca133f508964302effb3c9c2f209f85